Final/multinblkserver.cpp

```diff
@@ -0,0 +1,132 @@
+#include <unistd.h>
+#include <string.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <netdb.h>
+#include <iostream>
+#include "ThreadPool.h"
+#include "LinkedList.h"
+
+const int PORT_NO = 8080;
+const int BACKLOG_SIZE = 2;
+
+// Prototypes
+void handle_connection(fd_set read_fds, LinkedList *open_conn_fds, ThreadPool *thread_pool);
+void echo(int fd);
+void accept_connection(int sockfd, LinkedList *open_conn_fds);
+
+void error(const char *msg)
+{
+    perror(msg);
+    exit(0);
+}
+
+int main(int argc, char const *argv[])
+{
+    // Step 1: Set up socket
+    int sockfd = socket(AF_INET, SOCK_STREAM, 0); // Create socket file descriptor for server
+    if (sockfd < 0)
```

Comment:
nit: `if (socket(AF_INET, SOCK_STREAM, 0) < 0)`

Reply:
Don’t I need sockfd for later?

Reply:
ah yes my b